### PR TITLE
nodes_testcases

### PR DESCRIPTION
**Pull request by Dinaamagdy**

### CODE REVIEW

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py [hunk 1]
```diff
@@ -65,14 +73,21 @@ def test003_list_jobs(self):
...
-        self.lg.info('Compare results with golden value.')
 
+        self.lg.info('Compare results with golden value.')
+        jobs = response.json()
+        client_jobs = self.python_client.get_jobs_list()
+        self.assertEqual(len(jobs), len(client_jobs))
+        for job in jobs:
+            for client_job in client_jobs:
+                if job['id'] == client_job['id']:
+                    self.assertEqual(job['startTime'], client_job['starttime'])
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: there is no other elements to be compared plus 'startTime'?

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py [hunk 2]
```diff
@@ -107,60 +121,49 @@ def test005_get_job_details(self):
...
-        node_id = self.base_test.get_random_node()
-
         self.lg.info('Get list of jobs of this node .')
-        response = self.nodes_api.get_nodes_nodeid_jobs(node_id=node_id)
+        response = self.nodes_api.get_nodes_nodeid_jobs(node_id=self.node_id)
         self.assertEqual(response.status_code, 200)
 
         self.lg.info('Choose one of these jobs to list its details.')
         jobs_list = response.json()
         job_id = jobs_list[random.randint(0, len(jobs_list))]['id']
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: there is no jobs_list[len(job_list)] item. Please, set the randint limits to be from 0 to len(jobs_list)-1

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py [hunk 3]
```diff
@@ -171,17 +174,23 @@ def test007_ping_specific_node(self):
...
-        node_id = self.base_test.get_random_node()
-
         self.lg.info('post /nodes/{nodeid}/ping api.')
-        response = self.nodes_api.post_nodes_nodeid_ping(node_id=node_id)
+        response = self.nodes_api.post_nodes_nodeid_ping(node_id=self.node_id)
 
         self.lg.info('check response status code.')
         self.assertEqual(response.status_code, 200)
 
+        self.lg.info('check that you can get node, compare result with golden value. ')
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: Remove this step

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py [hunk 4]
```diff
@@ -313,21 +317,17 @@ def test013_get_nics_details(self):
...
-        # nics_info=response
-        # for nic_info,i in enumerate(nics_info):
-        #     for key in nic_info.keys:
-        #         self.assertEqual(nic_info[key],result[i][key])
-
+        result = self.python_client.get_nodes_nics()
+        nics_info = response.json()
+        for i, nic_info in enumerate(nics_info):
+            for key in nic_info.keys():
+                if key in result[i].keys():
```
Comment (islamTaha12) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: result[i] may not be matched with this nic_info order

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py (hunk 1 above)
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: yes , job list return id an start time only

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py [hunk 5]
```diff
@@ -21,21 +30,17 @@ def test001_list_nodes(self):
         #. Send get nodes api request.
         #. Compare results with golden value.
         """
-        self.lg.info('get nodes api request ')
+        self.lg.info('sendo get nodes api request ')
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: wrong 'o'

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py [hunk 6]
```diff
@@ -1,16 +1,25 @@
...
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.nodes_api = NodesAPI()
-        self.base_test = TestcasesBase()
 
     def setUp(self):
-        self.g8os_ip = self.config['g8os_ip']
+        self.node_id = self.get_random_node()
+        self.node = {}
+        for node in nodes:
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: done

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py (hunk 5 above)
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: done

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py [hunk 7]
```diff
@@ -21,21 +30,17 @@ def test001_list_nodes(self):
...
-        for node in self.utiles.nodes_info:
-            nodes_id.append(node['id'])
-
-        for node in data:
-            self.assertIn(node['id'], nodes_id)
+        Nodes_result = response.json()
+        self.assertEqual(len(Nodes_result), len(nodes))
+        for i, node in enumerate(Nodes_result):
+            for key in node.keys():
+                if key in nodes[i].keys():
```
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: done

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py (hunk 2 above)
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: done

File: functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py (hunk 3 above)
Comment (Dinaamagdy) [inline functional_testing/Grid_API_Testing/api_testing/testcases/basic_tests/test_nodeid_apis.py]: done

**Pull request merged**